lib/msal-node/test/network/HttpClient.spec.ts

```diff
@@ -195,53 +210,52 @@ describe("HttpClient", () => {
          * one for http timeout and one for the socket timeout inside of the socket
          */
         test("Via Proxy", async () => {
-            (http.request as jest.Mock).mockImplementationOnce(mockHttpRequest(mockPostResponseBody, statusCodeOk, statusCodeOk));
+            (http.request as jest.Mock).mockImplementationOnce(mockHttpRequest(mockPostResponseBody, proxyStatusCodeOk, socketStatusCodeOk));
             await expect(httpClient.sendPostRequestAsync(url, postNetworkRequestOptionsWithProxyUrl, timeoutInMilliseconds)).rejects.toEqual(error);
         });
     });
 
-
     describe("Bad Status Code Error", () => {
-        const statusCodeError: number = 500;
-        const error: Error = new Error("Error connecting to proxy");
-        const networkResponse: NetworkResponse<MockedMetadataResponse> = getNetworkResponse<MockedMetadataResponse>(mockGetResponseBody, statusCodeError);
+        const proxyError: Error = new Error("Error connecting to proxy. Status Code: 500. Status Message: Unknown");
 
         describe("Get Request", () => {
-            test("Via Https", async () => {
-                (https.request as jest.Mock).mockImplementationOnce(mockHttpsRequest(mockGetResponseBodyBuffer, statusCodeError));
-                await expect(httpClient.sendGetRequestAsync(url)).resolves.toEqual(networkResponse);
+            test("Via Https - Server Error 500", async () => {
```

Comment:
Are there any tests that cover the AAD error path, where error code is 400 or 401 and  `parsedBody = JSON.parse(body);` succeeds?

Reply:
There are not - the existing Http(s) 200 tests cover this. Even in the case you described above, the Http(s) request is successful (which is the purpose of the test). 

Consider an example where the user enters an incorrect client secret. The http(s) request will be successful, and the response's body from the AAD server will look like:
`{
  error: 'invalid_client',
  error_description: "AADSTS7000215: Invalid client secret provided. Ensure the secret being sent in the request is the client secret value, not the client secret ID, for a secret added to app '...'.\r\n" +
    'Trace ID: ...' +
    'Correlation ID: ...' +
    'Timestamp: ...',
  error_codes: [ 7000215 ],
  timestamp: '.',
  trace_id: '.',
  correlation_id: '.',
  error_uri: '...'
}`

The only difference between this and the existing Http(s) 200 tests is the status code and body... but we define those as constants.

Reply:
Can you please add a simple unit test for this. Status code 400 and the body being `{ error: 'invalid_client', error_description: "AADSTS7000215: Invalid client secret provided. Ensure the secret being sent in the request is the client secret value, not the client secret ID, for a secret added to app '...'.\r\n" + 'Trace ID: ...' + 'Correlation ID: ...' + 'Timestamp: ...', error_codes: [ 7000215 ], timestamp: '.', trace_id: '.', correlation_id: '.', error_uri: '...' }`